llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -411,6 +417,21 @@ std::string saveModuleIR(Module &M, int I, StringRef Suff) {
   return OutFilename;
 }
 
+bool isImportedFunction(const Function &F) {
+  // Functions with definitions are not imported
+  if (!F.isDeclaration())
+    return false;
+
+  bool ReturnValue = true;
+  if (auto NameStr = itaniumDemangle(F.getName())) {
+    StringRef DemangledName(NameStr);
+    if (DemangledName.starts_with("__") || DemangledName.starts_with("_spirv"))
```

Comment:
I'm not sure if we can universally assume anything with __ is not imported

Reply:
Hi @sarnex 
My understanding was from the design document which states this: "Declarations with __ prefix in demangled name are not recorded as imported functions". I am also not able to think of any scenarios where imported functions will start with "__". However, I could be missing something and hopefully our testing will reveal gaps, if any. 

Thanks

Reply:
Names that start with `__` are reserved in C++ for the compiler, meaning that if user defines function with such name, it is a UB. Therefore, the assumption is that any symbols that start with `__` are implementation-provided and such are not expected to come in a separate device image. At least for SYCL, we don't create any "utility" device images where we outline our own implementation-provided symbols.